library/Fission/Models.hs

```diff
@@ -0,0 +1,72 @@
+{-# OPTIONS_GHC -fno-warn-missing-export-lists #-}
+
+{-# LANGUAGE NoDeriveAnyClass #-}
+{-# LANGUAGE QuasiQuotes      #-}
+{-# LANGUAGE TemplateHaskell  #-}
+
+module Fission.Models where
+
+import           Database.Persist.Postgresql
+import           Database.Persist.TH
+
+import           Data.UUID
+import           Data.Swagger
+
+import           Network.IPFS.CID.Types
+
+import           Fission.Prelude
+
+import           Fission.Platform.Heroku.Region.Types
+import           Fission.Security
+import           Fission.User.Role.Types
+
+import           Fission.Internal.Orphanage.CID  ()
+import           Fission.Internal.Orphanage.UUID ()
+
+import qualified Fission.Internal.UTF8 as UTF8
+
+share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+HerokuAddOn sql=heroku_add_ons
+  uuid       UUID         sqltype=uuid
+  region     Region Maybe
+
+  insertedAt UTCTime
+  modifiedAt UTCTime
+
+  UniqueUUID uuid
+  deriving Show Eq
+
+User sql=users
+  username      Text
+  email         Text          Maybe
+  role          Role
+  active        Bool
+  herokuAddOnId HerokuAddOnId Maybe
+  secretDigest  SecretDigest
+
+  insertedAt    UTCTime
+  modifiedAt    UTCTime
+
+  UniqueUsername username
+  deriving Show Eq
+
+UserCID sql=user_cids
+  userFK     UserId
+  cid        CID
+
+  insertedAt UTCTime
+  modifiedAt UTCTime
+
+  UniqueUserCID userFK cid
+  deriving Show Eq
+|]
+
+instance Digestable UserId where
+  digest = digest . UTF8.textShow
+
+instance ToSchema UserId where
+  declareNamedSchema _ =
+    mempty
+      |> type_ ?~ SwaggerInteger
+      |> NamedSchema (Just "UserID")
+      |> pure
```

Comment:
Welcome to the big monolithic models module! We get some niceties by having it set up this way, including automatic migration generation, and it looks like the docs. We always have the option to break them up later. There's some work going into making this module more modular as part of Persistent. Today it's totally doable, but also a bit of a PITA.